src/components/FeatureViews/Accounts/ComponentViewer.js

```diff
@@ -1,81 +1,67 @@
+import React, {useMemo} from "react";
 import PropTypes from "prop-types";
-import React from "react";
 
-import Bio from "./TabComponents/Bio";
 import Grid from "@material-ui/core/Grid";
+
+import Bio from "./TabComponents/Bio";
 import InstructorCourses from "./TabComponents/InstructorCourses";
+import InstructorSchedule from "./TabComponents/InstructorSchedule";
 import Notes from "../Notes/Notes";
 import ParentContact from "./TabComponents/ParentContact";
 import PayCourses from "./TabComponents/PayCourses";
 import PaymentHistory from "./TabComponents/PaymentHistory";
-import InstructorSchedule from "./TabComponents/InstructorSchedule";
 import StudentCourseViewer from "./TabComponents/StudentCourseViewer";
 import StudentInfo from "./TabComponents/StudentInfo";
 
-const ComponentViewer = (props) => {
-    let component;
-
-    switch (props.inView) {
-        case 0:
-            component = <InstructorSchedule instructorID={props.user.user_id} />;
-            break;
-        case 1:
-            component = <InstructorCourses instructorID={props.user.user_id} />;
-            break;
-        case 2:
-            component = <Bio background={props.user.background} />;
-            break;
-        case 3:
-            component = (
-                <StudentCourseViewer
-                    current
-                    studentID={props.user.user_id} />
-            );
-            break;
-        case 4:
-            component = (
-                <StudentCourseViewer
-                    current={false}
-                    studentID={props.user.user_id} />
-            );
-            break;
-        case 5:
-            component = <PaymentHistory user_id={props.user.user_id} />;
-            break;
-        case 6:
-            component = <ParentContact parent_id={props.user.parent_id} />;
-            break;
-        case 7:
-            component = (
-                <Notes
-                    ownerID={props.user.user_id}
-                    ownerType={props.user.role} />
-            );
-            break;
-        case 8:
-            component = <StudentInfo user={props.user} />;
-            break;
-        case 9:
-            component = <PayCourses user={props.user} />;
-            break;
-        default:
-            component = <InstructorSchedule />;
-    }
+const ComponentViewer = ({inView, user}) => {
+    const component = useMemo(() => {
```

Comment:
We should change this to a JS object so that this isn't a big switch statement. This way we can keep adding new components to the components object more easily than adding a switch case